lib/librte_regexdev/rte_regexdev.c

```diff
@@ -2,4 +2,133 @@
  * Copyright(C) 2019 Marvell International Ltd.
  */
 
-#include <rte_regexdev.h>
+#include <string.h>
+
+#include <rte_spinlock.h>
+#include <rte_memory.h>
+#include <rte_memcpy.h>
+#include <rte_memzone.h>
+#include <rte_string_fns.h>
+
+#include "rte_regexdev.h"
+#include "rte_regexdev_driver.h"
+
+/* Shared memory between primary and secondary processes. */
+static struct {
+	uint64_t next_owner_id;
+	rte_spinlock_t ownership_lock;
+	struct rte_regex_dev_data data[RTE_MAX_REGEXDEV_PORTS];
+} *rte_regex_dev_shared_data;
+
+static const char *MZ_RTE_REGEX_DEV_DATA = "rte_regex_dev_data";
+
+struct rte_regex_dev rte_regex_devices[RTE_MAX_REGEXDEV_PORTS];
+
+/* spinlock for shared data allocation */
+static rte_spinlock_t rte_regex_shared_data_lock = RTE_SPINLOCK_INITIALIZER;
+
+
+static uint16_t
+regex_dev_find_free_port(void)
+{
+	unsigned i;
+
+	for (i = 0; i < RTE_MAX_REGEXDEV_PORTS; i++) {
+		if (rte_regex_devices[i].data != NULL &&
+		    rte_regex_devices[i].data->name == '\0')
```

Comment:
and only one check should be sufficient, either name or data.
no need to check two things to find free device.

Reply:
yes since the data is a pointer it can be null.